sdi/invoice.go

```diff
@@ -0,0 +1,65 @@
+package sdi
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/xml"
+	"fmt"
+)
+
+// SendInvoiceResponse defines the post invoice response structure
+type SendInvoiceResponse struct {
+	XMLName xml.Name `xml:"Envelope"`
+	Body    struct {
+		Response ReceiveFileResponse `xml:"rispostaSdIRiceviFile"`
+	} `xml:"Body"`
+}
+
+// InvoiceOpts defines the send invoice parameters
+type InvoiceOpts struct {
+	FileName string
+	FileBody []byte
+}
+
+// SendInvoice sends invoice content to SdI
+func SendInvoice(ctx context.Context, invOpts InvoiceOpts, c *Client, cfg Config) (*SendInvoiceResponse, error) {
+	soapEndpoint := cfg.SOAPReceiveFileEndpoint()
+
+	body := SoapRequestToSendInvoice(invOpts.FileName, invOpts.FileBody)
+	resp, err := c.Client.R().
+		SetBody(body).
+		SetContext(ctx).
+		Post(soapEndpoint)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.IsError() {
+		return nil, fmt.Errorf("response error: %v", resp)
+	}
+
+	response := &SendInvoiceResponse{}
+	err = ParseMultipartResponse(resp, response)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+// SoapRequestToSendInvoice prepares the request content for SOAP to send an invoice
+func SoapRequestToSendInvoice(fileName string, fileBody []byte) string {
```

Comment:
public?

Reply:
In this case I have a bit of a dilemma. I know this function doesn't have to be public, but I like that it only does one thing and I can easily test it.
Any ideas?